lib/src/ctx/context.rs

```diff
@@ -100,8 +111,7 @@ impl<'a> Context<'a> {
 	pub fn done(&self) -> Option<Reason> {
 		match self.deadline {
 			Some(deadline) if deadline <= Instant::now() => Some(Reason::Timedout),
-			// TODO: see if we can relax the ordering.
-			_ if self.cancelled.load(Ordering::SeqCst) => Some(Reason::Canceled),
+			_ if self.cancelled.load(Ordering::Relaxed) => Some(Reason::Canceled),
```

Comment:
I relaxed the memory ordering here because the memory ordering seems entirely irrelevant (for it to be relevant would probably require it to guard some `unsafe`, otherwise racy code), so it is not necessary to constrain it.